client/mm/mm_arb_market_maker.go

```diff
@@ -0,0 +1,642 @@
+package mm
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/client/mm/libxc"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/calc"
+	"decred.org/dcrdex/dex/order"
+)
+
+// ArbMarketMakingPlacement is the configuration for an order placement
+// on the DEX order book based on the existing orders on a CEX order book.
+type ArbMarketMakingPlacement struct {
+	Lots       uint64  `json:"lots"`
+	Multiplier float64 `json:"multiplier"`
+}
+
+// ArbMarketMakerConfig is the configuration for a market maker that places
+// orders on both sides of the DEX order book, at rates where there are
+// profitable counter trades on a CEX order book. Whenever a DEX order is
+// filled, the opposite trade will immediately be made on the CEX.
+//
+// The placements on the DEX order book will be made in the following way.
+// Suppose we have the following buy placements:
+//
+// 1. { Lots: 1, Multiplier: 1.5 }
+// 2. { Lots 1, Multiplier: 1.0 }
+//
+// The bot will find the rate at which it is possible to sell 1.5 lots on the
+// CEX, and put in a buy order for 1 lot at `rate / (1 + profit)`.
+// Then it will find the rate at which it is possible to sell 2.5 lots on the
+// CEX and put another buy order for 1 lot at `rate / (1 + profit)`.
+//
+// The first placement will be closest to the mid gap, and each subsequent one
+// will be farther from the mid-gap. The multiplier ensures that even if some
+// of the orders closest to the mid-gap are filled, there will still be enough
+// on the CEX orderbook for a profitable trade.
```

Comment:
Kinda seems like the `multiplier` is just extra padding that could be accomplished to a large degree by choosing a larger `profit` instead. Can you help me understand the `multiplier` knob better?

Reply:
To point of the `multiplier` is to avoid the situation where the trades on the CEX are matched before the bot can get around to making its counter-trade. `multiplier` gives the user more control over the size of the buffer than `profit`. Depending on how sparse the CEX orderbook is near the mid-gap, changing the `profit` could give you much less or much more buffer than you need. The point is to be able to lock in a certain amount of profit and have precise control over the probability you will be able to fill the counter-trade.